### PR TITLE
test/2375: Setup Cypress for E2E testing

### PR DESCRIPTION
**Pull request by Sidsector9**
### Description of the Change

The PR sets up Cypress to perform E2E testing.

#### Features
- [x] 	Autosuggest
- [x] 	Document
- [x] 	Protected Content
- [x] 	Related Posts
- [x] 	Search
- [x] 	WooCommerce

#### Indexables
- [x] 	User

#### General
- [x] 	Basic (these tests were moved to other locations)
- [x] 	Dashboard Sync
- [x] 	General
- [x] 	Highlighting (moved to the Search feature)
- [x] 	Standard (WP Basic Functionality)
- [x] 	WP-CLI

#### Post Tests
- [x] Clean up (deletion of indices)

### CODE REVIEW

**Comment by jeffpaul** [comment]
@felipeelia we should also be able to remove the WP Acceptance integration as part of this PR, correct?

**Comment by felipeelia** [comment]
yup @jeffpaul, that is the plan

File: tests/cypress/support/commands.js [hunk 1]
```diff
@@ -0,0 +1,310 @@
...
+		cy.get('.components-snackbar').should('be.visible');
+	}
+
+	/**
+	 * Give Elasticsearch some time to process the new post.
+	 *
+	 * @todo instead of waiting for an arbitrary time, we should ensure the post is stored.
+	 */
+	// eslint-disable-next-line cypress/no-unnecessary-waiting
+	cy.wait(2000);
```
Comment (Sidsector9) [inline tests/cypress/support/commands.js:146]: 👍

File: tests/cypress/integration/features/search/weighting.spec.js [hunk 2]
```diff
@@ -0,0 +1,63 @@
+describe('Post Search Feature - Weighting Functionality', () => {
+	it("Can't find a post by title it title is not marked as searchable", () => {
```
Comment (Sidsector9) [inline tests/cypress/integration/features/search/weighting.spec.js:2]: Just a minor typo here I think it should be `if` instead of `it`.

**Pull request merged**